src/pal/src/exception/signal.cpp

```diff
@@ -242,33 +237,59 @@ void SEHCleanupSignals()
 #if !HAVE_MACH_EXCEPTIONS
 /*++
 Function :
-    sigill_handler
+    invoke_previous_action
 
-    handle SIGILL signal (EXCEPTION_ILLEGAL_INSTRUCTION, others?)
+    synchronously invokes the previous action or aborts when that is not possible
 
 Parameters :
-    POSIX signal handler parameter list ("man sigaction" for details)
+    action  : previous sigaction struct
+    code    : signal code
+    siginfo : signal siginfo
+    context : signal context
+    signalRestarts: BOOL state : TRUE if the process will be signalled again
 
     (no return value)
 --*/
-static void sigill_handler(int code, siginfo_t *siginfo, void *context)
+static void invoke_previous_action(struct sigaction* action, int code, siginfo_t *siginfo, void *context, bool signalRestarts = true)
 {
-    if (PALIsInitialized())
+    _ASSERTE(action != NULL);
+
+    if (action->sa_handler == SIG_IGN)
```

Comment:
I am sorry, I was probably not clear. I meant that we should check the `action->sa_flags & SA_SIGINFO` first and never check the sa_handler for SIG_IDN / SIG_DFL if it is set.